pkg/sql/sql_activity_update_job.go

```diff
@@ -410,25 +434,46 @@ INTO system.public.transaction_activity
                     ts.fingerprint_id,
                     ts.agg_interval));
 `,
-		totalStmtClusterExecCount,
-		aggTs,
-		topLimit,
-	)
+			totalStmtClusterExecCount,
+			aggTs,
+			topLimit,
+		)
 
-	if err != nil {
 		return err
+	})
+
+	if errTxn != nil {
+		return errTxn
 	}
 
-	// Select the top 500 (controlled by sql.stats.activity.top.max) for each of
-	// execution_count, total execution time, service_latency, cpu_sql_nanos,
-	// contention_time, p99_latency. Also include all statements that are in the
-	// top N transactions. This is needed so the statement information is
-	// available for the ui so a user can see what is in the transaction.
-	_, err = u.db.Executor().ExecEx(ctx,
-		"activity-flush-stmt-transfer-tops",
-		nil, /* txn */
-		sessiondata.NodeUserSessionDataOverride,
-		`
+	errTxn = u.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+
+		// Delete all the rows of the old data from the table for the current
+		// aggregated timestamp. This is necessary because if a customer generates
+		// a lot of fingerprints each time the upsert runs it will add all new rows
+		// instead of updating the existing one. This causes the
+		// transaction_activity to grow to large causing the UI to be slow.
+		_, err := txn.ExecEx(ctx,
+			"activity-flush-txn-transfer-tops",
+			txn.KV(), /* txn */
+			sessiondata.NodeUserSessionDataOverride,
+			`DELETE FROM system.public.statement_activity WHERE aggregated_ts = $1;`,
```

Comment:
Curious, is this going to cause noticeable contentiom when we are attempting to read the sql activity tables via the UI and this job is modifying the table? Also do you think it would be worthwhile to run the stmt and txn table txns concurrently?